pydm/data_plugins/epics_plugins/pyepics_plugin_component.py

```diff
@@ -156,19 +156,19 @@ def remove_listener(self, channel):
         if channel.value_signal is not None:
             try:
                 channel.value_signal[str].disconnect(self.put_value)
-            except KeyError:
+            except (KeyError, TypeError):
```

Comment:
Why can't we just have a single try - catch for all these disconnections? If one of these disconnections has a problem, can we know which value_signal is causing the issue by looking at the call stack?

Reply:
Or have a disconnect() function that takes a signal and a slot, and have exception handling in one place (I have the same comment for the changes below).

Reply:
@hmbui that is a temporary fix to solve the problem described at #177 (which is an ancient issue) and now I have some clues to what it probably is... this is the easiest fix to get the app going. What I will do is to get back to that issue and fix properly by next week.